### PR TITLE
fix(text selector): allow single quoted text

### LINKED ISSUE #1882: [Feature] Support single quotes in text selector engine
To improve the developer experience I think, that it would help to also allow single quotes in the text selector engine due the fact that not all code bases use nowadays only single quotes as a standard.

If the code base uses double quotes, the user either has to escape them like that:

```js
await page.waitForSelector("text=\"fooobar\"")
```

or be inconsistent in this code base with quotes and the corresponding coding standard.

(if you agree with that, I'm open to submit a PR for this enhancement.

### PR DESCRIPTION
**Pull request by dgozman**
Fixes #1882.

### CODE REVIEW

File: src/injected/textSelectorEngine.ts [hunk 1]
```diff
@@ -48,10 +48,27 @@ export function createTextSelector(shadow: boolean): SelectorEngine {
   return engine;
 }
 
+function unescape(s: string): string {
+  if (s.indexOf('\\') === -1)
```
Comment (mxschmitt) [inline src/injected/textSelectorEngine.ts:52]: nit: I personally would use there .includes for readability

Comment (dgozman) [reply src/injected/textSelectorEngine.ts:52]: Good point!

File: src/injected/textSelectorEngine.ts [hunk 2]
```diff
@@ -48,10 +48,27 @@ export function createTextSelector(shadow: boolean): SelectorEngine {
...
+    r.push(s[i++]);
+  }
+  return r.join('');
+}
+
 type Matcher = (text: string) => boolean;
 function createMatcher(selector: string): Matcher {
-  if (selector[0] === '"' && selector[selector.length - 1] === '"') {
-    const parsed = JSON.parse(selector);
+  if (selector.length > 1 && selector[0] === '"' && selector[selector.length - 1] === '"') {
```
Comment (mxschmitt) [inline src/injected/textSelectorEngine.ts:66]: nit: both ifs here and in the selectors.ts are doing the same except for single and then for double quotes. I personally would use a smaller helper function for it.

**Review by mxschmitt — APPROVED** [review]
great work :)

**Review by pavelfeldman — APPROVED** [review]

**Pull request merged**